### PR TITLE
nextjs template

### PR DESCRIPTION
**Pull request by ivandimitrov8080**
a nextjs template with a dev shell and default build

### CODE REVIEW

**Comment by IogaMaster** [comment]
Thanks for contibuting! Would there be a way to get most of the hash info from the package-lock.json. The same way it's done in the Rust template with the Cargo.lock?
https://github.com/nix-community/templates/blob/c873911833b3e635ef11137ad9b7f42ea92b0b85/rust/flake.nix#L34-L36

File: nextjs/flake.nix [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        buildInputs = with pkgs; [
+          nodejs_20
+          openssl
+          nodePackages_latest.pnpm
+        ];
+        nativeBuildInputs = buildInputs;
+        npmDepsHash = "<prefetch-npm-deps package-lock.json>";
+      in
+      {
+        devShell = pkgs.mkShell {
```
Comment (IogaMaster) [inline nextjs/flake.nix]:
```suggestion
        devShells.default = pkgs.mkShell {
```

Comment (IogaMaster) [reply nextjs/flake.nix]: To fix flake error

File: nextjs/flake.nix [hunk 2]
```diff
@@ -0,0 +1,57 @@
+{
+  description = ''
```
Comment (IogaMaster) [inline nextjs/flake.nix]: Instead of describing what to do here. Add it to a welcome text in the root flake.nix

**Review by IogaMaster — CHANGES_REQUESTED** [review]

File: nextjs/flake.nix (hunk 2 above)
Comment (IogaMaster) [reply nextjs/flake.nix]: Like how its done here:

https://github.com/nix-community/templates/blob/c873911833b3e635ef11137ad9b7f42ea92b0b85/flake.nix#L20C1-L29

**Comment by ivandimitrov8080** [comment]
> Thanks for contibuting! Would there be a way to get most of the hash info from the package-lock.json. The same way it's done in the Rust template with the Cargo.lock?
> 
> https://github.com/nix-community/templates/blob/c873911833b3e635ef11137ad9b7f42ea92b0b85/rust/flake.nix#L34-L36

Unfortunately, I don't see how it can be done if possible. In [the documentation](https://nixos.org/manual/nixpkgs/stable/#javascript-tool-specific) it says to use `prefetch-npm-deps`, which works fine.

**Comment by IogaMaster** [comment]
Okay, I'll look into a solution. Thanks for your help!

**Pull request merged**